scripts/openmc-ace-to-hdf5

```diff
@@ -0,0 +1,205 @@
+                      if line.strip().endswith('+')]
+    for i in reversed(continue_lines):
+        lines[i] += lines[i].strip()[:-1] + lines.pop(i + 1)
+
+    # Create list of ACE libraries
+    for line in lines:
+        words = line.split()
+        if len(words) < 3:
+            continue
+
+        path = os.path.join(os.path.dirname(args.xsdir), words[2])
+        if path not in ace_libraries:
+            ace_libraries.append(path)
+
+elif args.xsdata is not None:
+    with open(args.xsdata, 'r') as xsdata:
+        for line in xsdata:
+            words = line.split()
+            if len(words) >= 9:
+                path = os.path.join(os.path.dirname(args.xsdata), words[8])
+                if path not in ace_libraries:
+                    ace_libraries.append(path)
+
+else:
+    ace_libraries = args.libraries
+
+nuclides = {}
+library = openmc.data.DataLibrary()
+
+for filename in ace_libraries:
+    # Check that ACE library exists
+    if not os.path.exists(filename):
+        warnings.warn("ACE library '{}' does not exist.".format(filename))
+        continue
+
+    lib = openmc.data.ace.Library(filename)
+    for table in lib.tables:
+        name, xs = table.name.split('.')
+        if xs.endswith('c'):
+            # Continuous-energy neutron data
+            if name not in nuclides:
+                try:
+                    neutron = openmc.data.IncidentNeutron.from_ace(
+                        table, args.metastable)
+                except Exception as e:
+                    print('Failed to convert {}: {}'.format(table.name, e))
+                    continue
+
+                print('Converting {} (ACE) to {} (HDF5)'.format(table.name,
+                                                                neutron.name))
+
+                # Determine filename
+                outfile = os.path.join(args.destination,
+                                       neutron.name.replace('.', '_') + '.h5')
+                neutron.export_to_hdf5(outfile, 'w', libver=args.libver)
+
+                # Register with library
+                library.register_file(outfile)
+
+                # Add nuclide to list
+                nuclides[name] = outfile
+            else:
+                # Then we only need to append the data
+                try:
+                    neutron = \
+                        openmc.data.IncidentNeutron.from_hdf5(nuclides[name])
+                    print('Converting {} (ACE) to {} (HDF5)'
+                          .format(table.name, neutron.name))
+                    neutron.add_temperature_from_ace(table, args.metastable)
+                    neutron.export_to_hdf5(nuclides[name] + '_1', 'w',
+                                           libver=args.libver)
+                    os.rename(nuclides[name] + '_1', nuclides[name])
+                except Exception as e:
+                    print('Failed to convert {}: {}'.format(table.name, e))
+                    continue
+
+        elif xs.endswith('t'):
```

Comment:
I would consider moving this to a function as `handleThermalScatteringData(...)` which makes the code modular, simpler to read, and allows the author to provide a function-level comment